qutip/__init__.py

```diff
@@ -63,10 +63,16 @@
     os.environ['KMP_DUPLICATE_LIB_OK'] = 'True'
 
 import platform
+import scipy
+from packaging import version
```

Comment:
I am adding an alias then.Here I was uncomfortable adding a scipy dependency just to get the version. But I did not find a way out of it.

Reply:
We already depend on scipy (otherwise why would you need the version check?).  The problems are that you've used `packaging` without specifying it as a dependency, and that the import could shadow the name `version`, which is a name we want exported to refer to the `qutip.version` module.

Reply:
Sorry I meant uncomfortable importing the whole package as it was not being importes anywhere in the file.

Reply:
`scipy` is well-designed, and just importing `scipy` is a very very light import.  You'd need to import every individual subpackage of `scipy` to cause performance issues (though actually not really, because `qutip` already imports a lot of the subpackges at initialisation, which we probably shouldn't).

Reply:
Iresolved the shadowing and I added packaging to `requrements.txt`, though as you mention it is the de facto stdlib. Should it be added somewhere else?

Reply:
`requirements.txt` doesn't actually have all that much to do with our specified requirements - it's usually just a convenience file (and one that we've historically forgotten to update).  The actual package requirements are in `setup.cfg`.

`packaging` is not required to have a functioning version of Python; it's not part of the Python standard library.  However, it's _such_ a common dependency (transitively - it's used by `setuptools`), that you already had it installed.  A brand-new, clean Python installation won't, though, so you have to specify it.

Reply:
I chose 18.1 as the minimal packaging release version based on the heuristic that it has to have at least an year of life. In any case I do not hold any strong opinions on this.

We should check the minimal version on the CI now that it has become a dependency right?

Reply:
The aim of version requirements is only to mark the minimum version that includes all the features we need, not to arbitrarily limit people.  As far as I'm aware, `packaging` has always had `packaging.version.parse`, so there's not need to specify a minimum version.  (Also my heuristic was we can't depend on anything younger than two years old at a minimum, but packaging 18.1 does satisfy that.)

Technically yeah, it needs to be included in the CI as well, though really it doesn't matter too much - it'll automatically be installed because our build process depends on `setuptools`.